src/mame/drivers/plygonet.cpp

```diff
@@ -418,13 +430,13 @@ void polygonet_state::main_map(address_map &map)
 	map(0x000000, 0x1fffff).rom();
 	map(0x200000, 0x21ffff).ram().w(m_palette, FUNC(palette_device::write32)).share("palette");
 	map(0x400000, 0x40001f).rw(m_k053936, FUNC(k053936_device::ctrl_r), FUNC(k053936_device::ctrl_w));
-	map(0x440000, 0x440fff).rw(FUNC(polygonet_state::polygonet_roz_ram_r), FUNC(polygonet_state::polygonet_roz_ram_w));
-	map(0x480000, 0x480003).r(FUNC(polygonet_state::polygonet_inputs_r));
-	map(0x4c0000, 0x4c0003).w(FUNC(polygonet_state::polygonet_sys_w));
-	map(0x500000, 0x503fff).ram().w(FUNC(polygonet_state::shared_ram_write)).share("shared_ram");
+	map(0x440000, 0x440fff).rw(FUNC(polygonet_state::roz_ram_r), FUNC(polygonet_state::roz_ram_w));
+	map(0x480000, 0x480003).r(FUNC(polygonet_state::inputs_r));
+	map(0x4c0000, 0x4c0003).w(FUNC(polygonet_state::sys_w));
+	map(0x500000, 0x503fff).ram().rw(FUNC(polygonet_state::shared_ram_read), FUNC(polygonet_state::shared_ram_write)).share("shared_ram");
 	map(0x504000, 0x504003).w(FUNC(polygonet_state::dsp_w_lines));
 	map(0x506000, 0x50600f).rw(FUNC(polygonet_state::dsp_host_interface_r), FUNC(polygonet_state::dsp_host_interface_w));
-	map(0x540000, 0x540fff).rw(FUNC(polygonet_state::polygonet_ttl_ram_r), FUNC(polygonet_state::polygonet_ttl_ram_w));
+	map(0x540000, 0x540fff).rw(FUNC(polygonet_state::ttl_ram_r), FUNC(polygonet_state::ttl_ram_w));
```

Comment:
Could you make `m_ttl_vram` a `required_shared_ptr` and use `.ram().share(m_ttl_vram).w(FUNC(polygonet_state::ttl_ram_w))` here?  It would avoid the need for the `ttl_ram_r` member and automatically save the content.

Reply:
Good call, both this and the one below. I'll push an update in a few minutes.